### PR TITLE
Cleanups for FIPS options..

### PR DESCRIPTION
**Pull request by slontis**
The options in fipsprov.c are now generated using macros with fips_indicator_params.inc. This should keep the naming consistent.

Some fips related headers have moved to providers/fips/include so that they can use fips_indicator_params.inc.
securitycheck.h now includes fipsindicator.h, and fipsindicator include fipscommon.h.

fipsinstall.c uses OSSL_PROV_PARAM_ for the configurable FIPS options rather than using OSSL_PROV_FIPS_PARAM_* as this was confusing as to which one should be used. fips_names.h just uses aliases now for existing public names.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man7/EVP_SIGNATURE-ED25519.pod
Comment (slontis) [inline doc/man7/EVP_SIGNATURE-ED25519.pod:59]: @paulidale - should this be no-verify-digested ??

File: doc/man7/EVP_ASYM_CIPHER-RSA.pod
Comment (nhorman) [inline doc/man7/EVP_ASYM_CIPHER-RSA.pod:82]: Can we change this name in a minor release?  I.e since OSSL_ASYM_CIPHER_PARAM_PKCS15_PADDING_DISABLED already exists would this be considered an ABI breakage?

File: doc/man7/EVP_SIGNATURE-ED25519.pod
Comment (paulidale) [reply doc/man7/EVP_SIGNATURE-ED25519.pod:59]: Yes

File: doc/man7/EVP_ASYM_CIPHER-RSA.pod
Comment (slontis) [reply doc/man7/EVP_ASYM_CIPHER-RSA.pod:82]: @paulidale  did you backport this particular name to other branches for testing purposes?

File: doc/man7/EVP_SIGNATURE-ED25519.pod
Comment (slontis) [reply doc/man7/EVP_SIGNATURE-ED25519.pod:59]: added no

**Comment by slontis** [comment]
Fixed the style checks for non macro lines.

File: doc/man7/EVP_ASYM_CIPHER-RSA.pod
Comment (paulidale) [reply doc/man7/EVP_ASYM_CIPHER-RSA.pod:82]: I'm pretty sure I didn't.  I don't see the define in the older branches.

Comment (slontis) [reply doc/man7/EVP_ASYM_CIPHER-RSA.pod:82]: I couldnt see it either, and this is a new item added just to master, so it should be ok to change.

File: include/openssl/fips_names.h
Comment (paulidale) [inline include/openssl/fips_names.h:44]: Should the names elsewhere for these be the new name?  E.g. in `fipsinstall.c` the old name is still being used.

**Review by paulidale — COMMENTED** [review]
Nice cleanup.

I'll have a deeper look next week.

File: include/openssl/fips_names.h
Comment (slontis) [reply include/openssl/fips_names.h:44]: These are aliases, so either is fine.. We cant do much with the header as it is public..

Comment (paulidale) [reply include/openssl/fips_names.h:44]: I figured it would be more consistent using the new throughout.

**Comment by slontis** [comment]
Looks like there is a intermittent test issue not related to this PR.

File: include/openssl/fips_names.h
Comment (FdaSilvaYY) [inline include/openssl/fips_names.h:41]: Typo: compatability

**Review by paulidale — COMMENTED** [review]
A thought: would it be worthwhile including the fipsinstall options as part of the .inc processing?  Keeps all the mutable stuff together.  It could also fold into the fipsinstall test.  Essentially taking a final cleansing step.

Okay to approve without that once the tests are passing.

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
> A thought: would it be worthwhile including the fipsinstall options as part of the .inc processing? Keeps all the mutable stuff together. It could also fold into the fipsinstall test. Essentially taking a final cleansing step.

For a future PR if we decide we do want this.

**Comment by paulidale** [comment]
Ping @openssl/committers for second review please.

**Review by mattcaswell — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**
